### PR TITLE
GH-28 Move to go.mod instead of glide

### PR DESCRIPTION
**Pull request by anon6789**
#### Summary
This pull requests replaces the vendor directory with go.mod. 
When using the `bot_sample.go` code in another project together with go.mod `go mod init/tidy` fails because of changed module path for mattermost-server/model (see [this](https://github.com/mattermost/mattermost-server/commit/2259b7f2a8b92b5dada097e77c284d5d14a01ba3) commit in mattermost-server).

Is a vendor directory still needed? 
Which go version should be supported? I used go 1.13 here.

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-bot-sample-golang/issues/28

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module github.com/mattermost/mattermost-bot-sample-golang
+
+go 1.13
+
+require (
+	github.com/Masterminds/glide v0.13.3 // indirect
```
Comment (hanzei) [inline go.mod]: Is this line needed any longer?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module github.com/mattermost/mattermost-bot-sample-golang
+
+go 1.13
```
Comment (hanzei) [inline go.mod]: Go 1.12 should be fine

File: go.mod (hunk 1 above)
Comment (anon6789) [reply go.mod]: i removed glide from go.mod

File: go.mod (hunk 2 above)
Comment (anon6789) [reply go.mod]: I've downgraded to Go 1.12

**Pull request merged**